routes/runtime/github.rb

```diff
@@ -5,5 +5,31 @@ class CloverRuntime
     if (runner = GithubRunner[vm_id: @vm.id]).nil? || (repository = runner.repository).nil?
       fail CloverError.new(401, "InvalidCredentials", "There was an error logging in")
     end
+
+    r.get "cache" do
+      fail CloverError.new(204, "NotFound", "No cache entry") if repository.access_key.nil?
+
+      keys, version = r.params["keys"]&.split(","), r.params["version"]
+      fail CloverError.new(400, "InvalidRequest", "Wrong parameters") if keys.nil? || keys.empty? || version.nil?
+
+      scopes = [runner.workflow_job&.dig("head_branch"), repository.default_branch].compact
```

Comment:
That's a good question. As you pointed out, this is documented in GitHub: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache. However, I tested a few examples and was unable to access the caches in branch A. I plan to test this further in case I missed something.

Even though the actual implementation might differ, we should adhere to GitHub's documentation. We don't have insight into the implementation details, but we do have the documentation. I didn't include this scope because we lack the necessary data. The workflow_job payload doesn't contain this information, and we need to find a way to obtain it. Some other providers transmit this information from the client to the backend, but since we can't fully trust client data, I prefer not to source the information from the client. Making an API call to the GitHub API introduces additional latency and dependency just for obtaining caches. We may begin to store extra data about workflow runs with additional webhook events, similar to what we do with GitHub repositories, but it's unclear how much data we should keep.

I've decided to handle this part in a future, more isolated PR and have opted for a more restrictive version as the initial release.